freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/backup/cloud/AdlsGen2BackupConfigGenerator.java

```diff
@@ -0,0 +1,63 @@
+package com.sequenceiq.freeipa.service.freeipa.backup.cloud;
+
+import java.nio.file.Paths;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.telemetry.fluent.cloud.AdlsGen2Config;
+
+@Component
+public class AdlsGen2BackupConfigGenerator extends CloudBackupConfigGenerator<AdlsGen2Config> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdlsGen2BackupConfigGenerator.class);
+
+    private static final String[] ADLS_GEN2_SCHEME_PREFIXES = {"abfs://", "abfss://"};
+
+    private static final String AZURE_DFS_DOMAIN_SUFFIX = ".dfs.core.windows.net";
+
+    private static final String AZURE_BLOB_STORAGE_SUFFIX = "blob.core.windows.net";
+
+    private static final String AZURE_BLOB_STORAGE_SCHEMA = "https://";
+
+    @Override
+    public String generateBackupLocation(String location, String clusterType,
+            String clusterName, String clusterId) {
+        AdlsGen2Config adlsGen2Config = generateBackupConfig(location);
+        String logFolder = resolveBackupFolder(adlsGen2Config, clusterType, clusterName, clusterId);
+        String hostPart = String.format("%s.%s", adlsGen2Config.getAccount(), AZURE_BLOB_STORAGE_SUFFIX);
+        String generatedLocation = String.format("%s%s", AZURE_BLOB_STORAGE_SCHEMA,
+                Paths.get(hostPart, adlsGen2Config.getFileSystem(), logFolder));
+        LOGGER.info("The following ADLS Gen2 base folder location is generated: {} (from {})",
+                generatedLocation, location);
+        return generatedLocation;
+    }
+
+    private AdlsGen2Config generateBackupConfig(String location) {
+        if (StringUtils.isNotEmpty(location)) {
+            boolean secure = location.startsWith(ADLS_GEN2_SCHEME_PREFIXES[1]);
+            String locationWithoutScheme = getLocationWithoutSchemePrefixes(location, ADLS_GEN2_SCHEME_PREFIXES);
+            String[] split = locationWithoutScheme.split("@");
+            String[] storageWithSuffix = split[0].split("/", 2);
+            String folderPrefix = storageWithSuffix.length < 2 ? "" :  "/" + storageWithSuffix[1];
+            if (split.length < 2) {
+                return new AdlsGen2Config(folderPrefix, storageWithSuffix[0], null, secure);
+            } else {
+                String[] splitByDomain = split[1].split(AZURE_DFS_DOMAIN_SUFFIX);
+                String account = splitByDomain[0];
+                if (splitByDomain.length > 1) {
+                    String folderPrefixAfterDomain = splitByDomain[1];
```

Comment:
so many magic numbers. why `2`?

Reply:
This is a direct copy of the Telemetry processing of the storage locations for logging.  Do we want to change this overall?

Reply:
then comes the question, why do we have to copy if it's the same?

Reply:
The functions and flow is not an outright copy. Right now it is using very similar logic on the parsing of the adls container and s3 buckets paths, but the functions are slightly different with the base backups folder being different and a changed call path.

As well, later on, we will be moving to a dedicated selection of the backup storage location to a stand-alone bucket/container but the UI and flow work was unable to be done in the time needed for that call. This allows for the separate parsing mechanism for the backup once it moves away from the telemetry storage location.